lib/Parse/ParseExpr.cpp

```diff
@@ -2949,6 +3063,9 @@ ExprResult Parser::DeferredParseBoundsExpression(std::unique_ptr<CachedTokens> T
   ConsumeAnyToken();   // Skip past the current token to the new tokens.
   ExprResult Result = ParseBoundsExpression();
 
+  if (StartsRelativeBoundsClause(Tok))
```

Comment:
Actually, I have figured out a simpler approach.   At line 3064, instead of calling ParseBoundsExpression, call ParseBoundsExpressionOrInteropType.    The root of the problem is that it is ParseBoundsExpressionOrInteropType that handle rel_align clause.